tap_airbyte/tap.py

```diff
@@ -732,8 +740,55 @@ def sync_all(self) -> None:
                 ):
                     self._process_log_message(airbyte_message)
                 elif airbyte_message["type"] == AirbyteMessage.STATE:
-                    state_message = airbyte_message["state"]
+                    # See: https://docs.airbyte.com/understanding-airbyte/database-data-catalog
+                    # for how this state should be handled.
+                    state_message = deepcopy(airbyte_message["state"])
                     state_type = state_message["type"]
+
+                    if "airbyte_state" not in self.airbyte_state:
+                        self.airbyte_state["airbyte_state"] = []
+
+                    # The airbyte_state_v2 here should adhere to the link above.
+                    existing_airbyte_state_v2: list[dict] = deepcopy(self.airbyte_state["airbyte_state"])
+                    if state_type == "STREAM":
```

Comment:
We check the possible `state_type` values twice, here and starting on line 794. Would it make sense to refactor this?

Reply:
I understand, I was trying to keep the logic separate intentionally since presumably we want to remove one at some point (the existing one). 

I can remove the old one too but I think we need it there for backwards compatibility? Or is there a version bump we can do.

Reply:
Yeah, Chesterton's fence says we probably don't want to touch the existing logic without confirmation that it continues working as before. Let's keep it.